network/src/main/java/org/jboss/as/network/OutboundSocketBinding.java

```diff
@@ -160,9 +160,6 @@ public String getUnresolvedDestinationAddress() {
      * @throws UnknownHostException If the destination address cannot be resolved
      */
     public synchronized InetAddress getResolvedDestinationAddress() throws UnknownHostException {
-        if (this.resolvedDestinationAddress != null) {
-            return this.resolvedDestinationAddress;
-        }
         this.resolvedDestinationAddress = InetAddress.getByName(this.unresolvedDestinationAddress);
         return this.resolvedDestinationAddress;
```

Comment:
These two lines should just be:

return InetAddress.getByName(this.unresolvedDestinationAddress);

So don't store the value if it wasn't provided to the constructor; re-resolve each time.

The resolvedDestinationAddress field should also be made final as it can no longer be set outside the constructor.

This and my comment above should address the problem Brad Maxwell mentioned in his comment.